app/controllers/phenotypes_controller.rb

```diff
@@ -3,22 +3,29 @@ class PhenotypesController < ApplicationController
   helper_method :sort_column, :sort_direction
 
   def index
-    @title = "Listing all phenotypes"
-    @phenotypes = Phenotype.order(sort_column + " " + sort_direction)
-    @phenotypes_paginate = @phenotypes.paginate(:page => params[:page],:per_page => 10)
-    @phenotypes_json = []
-    @phenotypes.each do |p|
-      @phenotype = {}
-      @phenotype["id"] = p.id
-      @phenotype["characteristic"] = p.characteristic
-      @phenotype["known_variations"] = p.known_phenotypes
-      @phenotype["number_of_users"] = p.user_phenotypes.length
-      @phenotypes_json << @phenotype
-    end
+    @title = 'Listing all phenotypes'
+
+    @phenotypes = Phenotype
+      .order(sort_column + ' ' + sort_direction)
+      .includes(:user_phenotypes)
+
+    @phenotypes_paginate = @phenotypes.paginate(:page => params[:page], :per_page => 10)
+
     respond_to do |format|
       format.html
       format.xml 
-      format.json {render :json => @phenotypes_json} 
+      format.json do
+        phenotypes_json =
+          @phenotypes.find_each.map do |p|
+            phenotype = {}
+            phenotype['id'] = p.id
+            phenotype['characteristic'] = p.characteristic
+            phenotype['known_variations'] = p.known_phenotypes
+            phenotype['number_of_users'] = p.user_phenotypes.length
```

Comment:
I was just thinking that this would get fixed when the switch to jbuilder is made.